setup.py

```diff
@@ -35,6 +35,8 @@
                 "pygccxml.declarations",
                 "pygccxml.parser",
                 "pygccxml.utils"],
+      install_requires=[
+          "castxml>=0.3.6"],
```

Comment:
Does this mean that pygccxml would not be backward compatible anymore?

I guess that the castxml_epic_version was not bumped in castxml as this was just an addition of new content? Can we make sure that pygccxml still works with older versions of castxml?

Reply:
@iMichka, I will be removing that line.  I thought that was how one updated the version used in the testing which was proven false later.  

At this point, there shouldn't be any changes in backwards compatibility.  I will revisit how "no comments" are dealt with if we use an older version.
Thank you for catching it.

Reply:
How backwards compatible does pygccxml need to be with castxml? I don't think it makes sense to run a really old castxml with a new pygccxml.  The two seem like a pair and castxml is pip installable so they can move in lock step with the pip install eco system.

Reply:
Per VC, would be to ensure that lock step also is done in XML parsing too. (Prolly obvious, but just wanna state it.)

Reply:
Since I took over the project, I tried to be as much backward compatible as possible. Of course I would not expect us to support very old castxml versions. It's quite hard to keep both in sync, and I would personnaly recommend people to always use the latest version of both tools. But this is not always possible.
If we introduce a back-ward incompatible change in pygccxml we will also have to bump the major version.

I think adding support for comments should/can be done without breaking support for XML files without comments.

Reply:
My resolution of backwards compatibility was to ensure that the default value for comments was the same as a found comment instead of returning none.   I'm going to mark this as resolved.